### DISCUSSION TITLE
Eigen is required ?

### DISCUSSION THREAD

**Discussion by flaviu22** [cnr-isti-vclab/meshlab#867] [Q&A] [answered]
I get a fresh meshlab repository on my local machine. And when I want to compile it, I got the following message:

```
  Could not find a package configuration file provided by "Eigen3" with any
  of the following names:
    Eigen3Config.cmake
    eigen3-config.cmake

```
I need to install this Eigen component ?

**Comment by flaviu22** [comment]
One more thing: the same is for **GLEW**:

`Could NOT find GLEW (missing: GLEW_INCLUDE_DIRS GLEW_LIBRARIES) 
`
even if I saw this component on D:\Project\meshlab\src\external\glew-2.1.0 ... 

I checked _CMakeLists.txt_ regarding that:

```
message(STATUS "Searching for required components with bundled fallback")
find_package(GLEW)
find_package(Eigen3)

```
I should manually setup somewhere this paths ? I am using **Qt Creator** only where I opened **CMakeLists.txt** from **src** folder.

**Comment by alemuntoni** [comment]
Both eigen and glew are not required to be installed in the system.
The build system will first look if they are installed, and if they are not, it falls back on the bundled libraries inside the meshlab repository.
Therefore meshlab should build correctly also if CMake is displaying the messages you wrote!

**Comment by flaviu22** [comment]
Thank you for your time and patience. Something is puzzle me. I have the right configuration, I guess:

![image](https://user-images.githubusercontent.com/37621568/102236770-90fb4880-3efc-11eb-9fe3-5c20682020fb.png)

![image](https://user-images.githubusercontent.com/37621568/102236795-99ec1a00-3efc-11eb-87d2-f3758cbd0c37.png)

but when I try to run CMakeLists.txt from src folder:

```
Running C:\Program Files\CMake\bin\cmake.exe -S D:/Project/meshlab/src -B C:/Users/Flaviu/AppData/Local/Temp/QtCreator-MeJgjp/qtc-cmake-TSULdVbe in C:\Users\Flaviu\AppData\Local\Temp\QtCreator-MeJgjp\qtc-cmake-TSULdVbe.
-- Searching for required components
-- Searching for required components with bundled fallback
-- Could NOT find GLEW (missing: GLEW_INCLUDE_DIRS GLEW_LIBRARIES) 
CMake Warning at CMakeLists.txt:35 (find_package):
  By not providing "FindEigen3.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Eigen3", but
  CMake did not find one.
  Could not find a package configuration file provided by "Eigen3" with any
  of the following names:
    Eigen3Config.cmake
    eigen3-config.cmake
  Add the installation prefix of "Eigen3" to CMAKE_PREFIX_PATH or set
  "Eigen3_DIR" to a directory containing one of the above files.  If "Eigen3"
  provides a separate development package or SDK, be sure it has been
  installed.
-- Building with single precision
CMake Error at external/eigen.cmake:17 (message):
  Eigen is required - at least one of ALLOW_SYSTEM_EIGEN or
  ALLOW_BUNDLED_EIGEN must be enabled and found.
Call Stack (most recent call first):
  external/external_common.cmake:16 (include)
  CMakeLists.txt:125 (include)
-- Configuring incomplete, errors occurred!
See also "C:/Users/Flaviu/AppData/Local/Temp/QtCreator-MeJgjp/qtc-cmake-TSULdVbe/CMakeFiles/CMakeOutput.log".
CMake process exited with exit code 1.
Elapsed time: 00:01.

```
but something went wrong, because I cannot built it:

![image](https://user-images.githubusercontent.com/37621568/102238023-f996f500-3efd-11eb-8fe1-daf430dfcfeb.png)


Strange ...

**Comment by alemuntoni** [comment, marked as answer]
Just to be sure: did you clone the repo using the `--recursive` flag? Could you check if the `vcglib` directory is present inside `src`?

**Comment by flaviu22** [comment]
Sorry for late response, yes, there is a **vcglib** folder, but is empty:

![image](https://user-images.githubusercontent.com/37621568/102255311-906dac80-3f12-11eb-8ab2-0609e678f7ab.png)

I will try to do another clone and giveyou feedback. Thank you so much !!!

**Reply by flaviu22** [reply]
Yes, is compiled fine if I retrieve the code with --recursive flag.
